core/txpool/validation.go

```diff
@@ -101,25 +101,23 @@ func ValidateTransaction(tx *types.Transaction, head *types.Header, signer types
 	if tx.GasFeeCapIntCmp(tx.GasTipCap()) < 0 {
 		return core.ErrTipAboveFeeCap
 	}
-	if tx.Type() != types.DepositTxType {
-		// Make sure the transaction is signed properly
-		if _, err := types.Sender(signer, tx); err != nil {
-			return ErrInvalidSender
-		}
-		// Ensure the transaction has more gas than the bare minimum needed to cover
-		// the transaction metadata
-		intrGas, err := core.IntrinsicGas(tx.Data(), tx.AccessList(), tx.To() == nil, true, opts.Config.IsIstanbul(head.Number), opts.Config.IsShanghai(head.Number, head.Time), false)
-		if err != nil {
-			return err
-		}
-		if tx.Gas() < intrGas {
-			return fmt.Errorf("%w: needed %v, allowed %v", core.ErrIntrinsicGas, intrGas, tx.Gas())
-		}
-		// Ensure the gasprice is high enough to cover the requirement of the calling
-		// pool and/or block producer
-		if tx.GasTipCapIntCmp(opts.MinTip) < 0 {
-			return fmt.Errorf("%w: tip needed %v, tip permitted %v", ErrUnderpriced, opts.MinTip, tx.GasTipCap())
-		}
+	// Make sure the transaction is signed properly
```

Comment:
Removed the check for deposit tx types here because they generally shouldn't be going through this flow.